xwiki-platform-core/xwiki-platform-bridge/src/main/java/org/xwiki/bridge/event/UserAuthenticatedEvent.java

```diff
@@ -0,0 +1,56 @@
+/*
+ * See the NOTICE file distributed with this work for additional
+ * information regarding copyright ownership.
+ *
+ * This is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU Lesser General Public License as
+ * published by the Free Software Foundation; either version 2.1 of
+ * the License, or (at your option) any later version.
+ *
+ * This software is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this software; if not, write to the Free
+ * Software Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA
+ * 02110-1301 USA, or see the FSF site: http://www.fsf.org.
+ */
+package org.xwiki.bridge.event;
+
+import org.xwiki.model.reference.DocumentReference;
+
+/**
+ * An event triggered when a user is authenticated.
+ * <p>
+ * The event also send the following parameters:
+ * </p>
+ * <ul>
+ * <li>source: the current {com.xpn.xwiki.user.api.XWikiUser} instance which has been authenticated.</li>
```

Comment:
Really not a fan of this parameter. `com.xpn.xwiki.user.api.XWikiUser` hardly a reference these days to represent a user in Java. Also, it's located in xwiki-platform-oldcore and xwiki-platform-bridge does not depend on it (that's actually the whole point of xwiki-platform-bridge to not depending on xwiki-platform-oldcore).

It would probably make much more sense to put this event in xwiki-platform-user-api and pass nothing as parameter since you already have the reference in the event itself.

Reply:
> It would probably make much more sense to put this event in xwiki-platform-user-api and pass nothing as parameter since you already have the reference in the event itself. It would increase a lot the number of modules which could use this event. I updated the jira issue description which was very old.

Yes I mentioned this in comments too but I'm not sure since authentication has its own module so right now it feels better in the security module than in the user module to me.

Reply:
I got your different points of view.

So far I'm trying to implement in through authentication module in security since we're talking about authentication but does the event still have to be sent in oldcore ?

Is it the only place where the user is authenticated ?

Reply:
> does the event still have to be sent in oldcore ?

Sure. What produces the event and where the event is defined are two different things.

Reply:
Of course, just wanted to be sure of that. Thanks!